### PR TITLE
try to add andoird and iphone support to vulkan-hpp

### PR DESCRIPTION
**Pull request by Latias94**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/v/vulkan-hpp/xmake.lua [hunk 1]
```diff
@@ -17,7 +17,7 @@ package("vulkan-hpp")
         add_extsources("pacman::vulkan-headers")
     end
 
-    on_install("windows|x86", "windows|x64", "linux", "macosx", "mingw", function (package)
+    on_install("windows|x86", "windows|x64", "linux", "macosx", "mingw", "android", "iphoneos", function (package)
         local arch_prev
         if package:is_plat("mingw") and package.plat_set then
```
Comment (SirLynix) [inline packages/v/vulkan-hpp/xmake.lua:22]: Try to replace by `if (package:is_plat("mingw") or package:is_cross()) and package.plat_set then`

**Comment by Latias94** [comment]
```shell
> clang "-fPIC" "-Qunused-arguments"
> /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -c -Qunused-arguments -std=c++14 -isystem /Users/runner/.xmake/packages/v/vulkan-hpp/v1.3.231/12e9eedf014f47a9b94cc461b16f47af/include -o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_CC5B46881E20490081F43D5909FA2020.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_3925859903DD4745864433957D53AB64.cpp
> checking for c++ includes(vulkan/vulkan.hpp)
> checking for c++ snippet(test)
checkinfo: ...amdir/core/sandbox/modules/import/core/tool/compiler.lua:84: @programdir/modules/core/tools/gcc.lua:718: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/wchar.h:123:15: fatal error: 'wchar.h' file not found
#include_next <wchar.h>
              ^~~~~~~~~
1 error generated.
  > in /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_3925859903DD4745864433957D[53](https://github.com/xmake-io/xmake-repo/actions/runs/4455140784/jobs/7824768505?pr=1926#step:4:54)AB64.cpp
```
How to resolve this error? 
Something like the code below?
```lua
add_csnippets("wchar.h", [[#include <wchar.h>
wchar_t w;]])
```

**Comment by waruqi** [comment]
> ```shell
> > clang "-fPIC" "-Qunused-arguments"
> > /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -c -Qunused-arguments -std=c++14 -isystem /Users/runner/.xmake/packages/v/vulkan-hpp/v1.3.231/12e9eedf014f47a9b94cc461b16f47af/include -o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_CC5B46881E20490081F43D5909FA2020.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_3925859903DD4745864433957D53AB64.cpp
> > checking for c++ includes(vulkan/vulkan.hpp)
> > checking for c++ snippet(test)
> checkinfo: ...amdir/core/sandbox/modules/import/core/tool/compiler.lua:84: @programdir/modules/core/tools/gcc.lua:718: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/wchar.h:123:15: fatal error: 'wchar.h' file not found
> #include_next <wchar.h>
>               ^~~~~~~~~
> 1 error generated.
>   > in /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/.xmake501/230318/_3925859903DD4745864433957D[53](https://github.com/xmake-io/xmake-repo/actions/runs/4455140784/jobs/7824768505?pr=1926#step:4:54)AB64.cpp
> ```
> 
> How to resolve this error? Something like the code below?
> 
> ```lua
> add_csnippets("wchar.h", [[#include <wchar.h>
> wchar_t w;]])
> ```

We need fix prev platform.

```diff
        local arch_prev
+      local plat_prev
        if (package:is_plat("mingw") or package:is_cross()) and package.plat_set then
            arch_prev = package:arch()
+           plat_prev = package:plat()
            package:plat_set(os.host())
            package:arch_set(os.arch())
        end
        import("package.tools.cmake").build(package, {buildir = "build"})
        if arch_prev and plat_prev then
-           package:plat_set("mingw")
+           package:plat_set(plat_prev)
            package:arch_set(arch_prev)
        end
```

**Comment by Latias94** [comment]
All checks pass now!

**Pull request merged**